### PR TITLE
Duplicate microchip fix2

### PR DESCRIPTION
**Pull request by jbarnsley10**

### CODE REVIEW

File: app/routes/cdo/create/microchip-search.js [hunk 1]
```diff
@@ -75,3 +78,15 @@ const duplicateMicrochipsInSession = (request, details) => {
...
+
+const getLatestDogIndex = (request) => {
+  if (getDogs(request)) {
+    return getDogs(request)?.filter(x => x.applicationType).length + 1
+  }
+
+  return 1
+}
+
+const determineDogId = (request, details) => {
```
Comment (whitewaterdesign) [inline app/routes/cdo/create/microchip-search.js:90]: not one for this PR, but would probably be good to refactor this a bit so logic is clearer

File: app/routes/cdo/create/microchip-search.js [hunk 2]
```diff
@@ -53,19 +56,19 @@ module.exports = [{
...
     }
   }
 }]
 
 const duplicateMicrochipsInSession = (request, details) => {
   const dogs = getDogs(request)
   if (dogs) {
-    dogs.forEach((d, ind) => { d.id = ind + 1 })
-    const dupeDogs = dogs.filter(d => d.microchipNumber === details.microchipNumber)
+    dogs.forEach((d, ind) => { d.dogId = d.dogId || `${ind + 1}` })
```
Comment (whitewaterdesign) [inline app/routes/cdo/create/microchip-search.js:67]: is there a test to check for the `${d.dogId}` !== `${details.dogId}` case?

**Review by whitewaterdesign — APPROVED** [review]
may or may not be tested - the new logic, but would be good to have a unit test on it

File: app/routes/cdo/create/microchip-search.js (hunk 2 above)
Comment (jbarnsley10) [reply app/routes/cdo/create/microchip-search.js:67]: Explicitly added now

**Pull request merged**